user/super/com/google/gwt/emul/java/util/Objects.java

```diff
@@ -129,6 +129,37 @@ public static <T> T requireNonNull(T obj, Supplier<String> messageSupplier) {
     return obj;
   }
 
+  public static <T> T requireNonNullElse(T obj, T defaultObj) {
+    return obj != null ? obj : requireNonNull(defaultObj);
+  }
+
+  public static <T> T requireNonNullElseGet(T obj, Supplier<? extends T> supplier) {
+    return obj != null ? obj : requireNonNull(supplier.get());
+  }
+
+  public static int checkIndex(int index, int length) {
+    if (index < 0 || index >= length) {
+      new IndexOutOfBoundsException("Index " + index + " out of bounds for length " + length);
```

Comment:
Missing a throw here
```suggestion
      throw new IndexOutOfBoundsException("Index " + index + " out of bounds for length " + length);
```